intra_process_demo/CMakeLists.txt

```diff
@@ -127,18 +116,22 @@ if(AMENT_ENABLE_TESTING)
 
       set(RCLCPP_DEMOS_EXECUTABLE "")
       foreach(executable ${demo_executables})
-        list(APPEND RCLCPP_DEMOS_EXECUTABLE "${CMAKE_CURRENT_BINARY_DIR}${test_executable_subfolder}/${executable}${target_suffix}${test_executable_extension}")
+        list(APPEND RCLCPP_DEMOS_EXECUTABLE "$<TARGET_FILE:${executable}${target_suffix}>")
       endforeach()
 
       string(REPLACE ";" "_" exe_list_underscore "${demo_executables}")
       configure_file(
         test/test_executables_demo.py.in
-        test_${exe_list_underscore}${target_suffix}.py
+        test_${exe_list_underscore}${target_suffix}.py.configured
         @ONLY
       )
+      file(GENERATE
+        OUTPUT "test_${exe_list_underscore}${target_suffix}_$<CONFIGURATION>.py"
+        INPUT "${CMAKE_CURRENT_BINARY_DIR}/test_${exe_list_underscore}${target_suffix}.py.configured"
+      )
 
       ament_add_nose_test(test_demo_${exe_list_underscore}${target_suffix}
-        "${CMAKE_CURRENT_BINARY_DIR}/test_${exe_list_underscore}${target_suffix}.py"
+        "${CMAKE_CURRENT_BINARY_DIR}/test_${exe_list_underscore}${target_suffix}_$<CONFIGURATION>.py"
```

Comment:
That's really stupid of them to deprecate a value and move to a new value that had a completely different meaning in their last major version.